src/main/java/nextstep/courses/domain/session/Session.java

```diff
@@ -0,0 +1,55 @@
+package nextstep.courses.domain.session;
+
+import nextstep.courses.domain.strategy.PaymentStrategy;
+import nextstep.payments.domain.Payment;
+
+import java.time.LocalDate;
+import java.util.Objects;
+
+public class Session {
+
+    private final SessionImage sessionImage;
+    private final SessionState sessionState;
+    private final PaymentStrategy sessionStrategy;
+    private final SessionDate sessionDate;
+
+    public Session(int size, String imageType, int width, int height, SessionState sessionState, PaymentStrategy sessionStrategy, LocalDate startDate, LocalDate endDate) {
+        this(new SessionImage(size, imageType, width, height), sessionState, sessionStrategy, new SessionDate(startDate, endDate));
+    }
+
+    public Session(SessionImage sessionImage, SessionState sessionState, PaymentStrategy sessionStrategy, SessionDate sessionDate) {
+        this.sessionImage = sessionImage;
+        this.sessionState = sessionState;
+        this.sessionStrategy = sessionStrategy;
+        this.sessionDate = sessionDate;
+    }
+
+    public boolean applyForCourse(Payment payment, LocalDate date) {
+        if (!SessionState.isOpen(this.sessionState)) {
```

Comment:
```suggestion
        if (!this.sessionState.canEnroll()) {
```

위 처럼 메세지를 구체화하는건 어떨까요?

Reply:
오! 입력 파라미터를 줄이는 좋은 방법이네요 ! 감사합니다 :)